### PR TITLE
Add promotion model

### PR DESCRIPTION
**Pull request by nvoers**
### Summary
Adds a new app for event organizers to request promotion material/messages. 

### How to test
1. Create promo requests for events via the events admin
2. Create promo requests via the 'regular' admin
3. Emails should be sent after creating a request
4. No emails are sent on updating an event (might be added later if desirable)

### CODE REVIEW

File: website/promotion/models.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    def __str__(self):
+        return "Promotion request for " + str(self.event)
+
+    def save(self, **kwargs):
+        if not self.publish_date:
+            self.publish_date = self.event.start.date()
+        notify_new_request(self)
+
+        return super().save(kwargs)
```
Comment (JobDoesburg) [inline website/promotion/models.py:58]: Add a meta class to give translatable model names, plural names and possibly think about custom permissions (default permissions are `add` `change` `delete` and `view` but maybe more advanced permissions are required...)

File: website/promotion/templates/requests/new_request_email.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{% load i18n %}{% blocktrans with request_event_title=request.event.title request_publish_date=request.publish_date request_channel=request.channel%}Hi Paparazcie,
```
Comment (JobDoesburg) [inline website/promotion/templates/requests/new_request_email.txt:1]: I don't think this `with (...)` is required here as it's not doing anything special

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,594 @@
+argon2-cffi==21.1.0; python_version >= "3.5" \
```
[the remark this answers is not in the document]
Comment (nvoers) [reply requirements.txt:1]: removed the file

File: website/events/services.py [hunk 4]
```diff
@@ -128,6 +129,19 @@ def is_organiser(member, event):
     return False
 
 
+def is_editor(member):
+    if member and member.is_authenticated:
+        if member.is_superuser or member.has_perm("promotion_requests.edit_status"):
+            return True
+        
+        member_in_paparazcie = member.get_member_groups().filter(pk=MemberGroup.objects.filter(name = "Paparazcie").pk).count() != 0
```
[the remark this answers is not in the document]
Comment (se-bastiaan) [reply website/events/services.py:137]: +1

File: website/promotion/models.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    )
+    channel = models.ForeignKey(
+        PromotionChannel,
+        verbose_name = _("channel"),
+        on_delete=models.CASCADE
+    )
+    assigned_to = models.CharField(
+        max_length = 40,
+        verbose_name=_("Assigned to"),
+        default = ""
```
Comment (JobDoesburg) [inline website/promotion/models.py:43]: I would make this field `null=True, blank=True`

File: website/promotion/models.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    )
+
+    status = models.CharField(
+        max_length = 40,
+        choices = STATUS_TYPES,
+        verbose_name = _("status"),
+        default = NOT_STARTED,
+    )
+    drive_folder = models.URLField(
+        max_length = 200,
```
Comment (JobDoesburg) [inline website/promotion/models.py:65]: I would make this field `null=True, blank=True`

Comment (JobDoesburg) [reply website/promotion/models.py:65]: Also I think 200 chars is too low, google drive urls can get pretty long

File: website/promotion/models.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        default = ""
+    )
+    remarks = HTMLField(("remarks"),)
```
Comment (JobDoesburg) [inline website/promotion/models.py:68]:
```suggestion
    remarks = HTMLField(verbose_name=_("remarks"),)
```

File: website/promotion/models.py [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    created_at = models.DateTimeField(
```
Comment (JobDoesburg) [inline website/promotion/models.py:69]:
```suggestion
    created_at = models.DateTimeField(
    verbose_name=_("created_at"),
```

File: website/promotion/migrations/0006_auto_20211215_1201.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+            model_name='promotionrequest',
+            name='publish_date',
+            field=models.DateField(default=datetime.date.today, verbose_name='Publish date'),
+        ),
+        migrations.AlterField(
+            model_name='promotionrequest',
+            name='status',
+            field=models.CharField(choices=[('not_started', 'Not started'), ('started', 'Started'), ('finished', 'Finished'), ('published', 'Published')], default='not_started', max_length=40, verbose_name='status'),
+        ),
+    ]
```
Comment (JobDoesburg) [inline website/promotion/migrations/0006_auto_20211215_1201.py:36]: When all model changes are done, please merge all migrations (remove all of them, re-make them)

**Review by JobDoesburg — CHANGES_REQUESTED** [review]
Can you edit your PR to allow other maintainers to access your branch? I currently cannot checkout your code to work on it (should be a checkbox somewhere on the right)

**Comment by JobDoesburg** [comment]
@nvoers sorry I had to rebase and became author of your commits :)

File: website/promotion/templates/requests/new_request_email.txt [hunk 10]
```diff
@@ -0,0 +1,12 @@
+{% load i18n bleach_tags %}{% blocktrans %}Hi Paparazcie,
```
Comment (se-bastiaan) [inline website/promotion/templates/requests/new_request_email.txt:1]:
```suggestion
{% load i18n bleach_tags %}{% blocktrans %}Hi,
```

**Review by se-bastiaan — DISMISSED** [review]

**Review by JobDoesburg — APPROVED** [review]

**Comment by pingiun** [comment]
> @nvoers sorry I had to rebase and became author of your commits :)

A rebase shouldn't change the author, just the committer

**Comment by JobDoesburg** [comment]
> > @nvoers sorry I had to rebase and became author of your commits :)
> 
> A rebase shouldn't change the author, just the committer

Then I don't know how this happened hahaha

**Comment by pingiun** [comment]
> > > @nvoers sorry I had to rebase and became author of your commits :)
> > 
> > 
> > A rebase shouldn't change the author, just the committer
> 
> Then I don't know how this happened hahaha

It's correct:
image

**Review by se-bastiaan — APPROVED** [review]

**Pull request merged**